packages/extensions/openapi-to-typespec/src/generate/generate-operations.ts

```diff
@@ -59,6 +71,55 @@ export function generateOperation(operation: TypespecOperation, operationGroup?:
   return statements.join("\n");
 }
 
+export function generateProviderAction(operation: TspArmProviderActionOperation) {
+  const doc = generateDocs(operation);
+  const summary = generateSummary(operation);
+  const statements: string[] = [];
+  summary && statements.push(summary);
+  statements.push(doc);
+  const templateParameters = [];
+  const responses = [...new Set(operation.responses)];
+  // Workaround for array response, refactor later.
+  const response =
+    responses.length === 1 && responses[0].endsWith("[]") ? `{@body _: ${responses[0]}}` : responses.join("|");
```

Comment:
what if it is not array?

Reply:
Then there will not have the `{@body}` wrapper.